src/yz_solrq_sup.erl

```diff
@@ -20,70 +20,112 @@
 
 -behaviour(supervisor).
 
--export([start_link/0, start_link/2, start_drain_fsm/1, child_count/1]).
+-export([start_link/0, start_drain_fsm/1, child_count/1, start_queue_pair/2, active_queues/0, sync_active_queue_pairs/0]).
 
 -include("yokozuna.hrl").
 
 -export([init/1]).
 
-%% used only by yz_perf on resize (manual operation)
--export([child_spec/2]).
 
 %%%===================================================================
 %%% API functions
 %%%===================================================================
 
 start_link() ->
-    start_link(worker_procs(), helper_procs()).
-
-start_link(NumQueues, NumHelpers) ->
-    supervisor:start_link({local, ?MODULE}, ?MODULE, [NumQueues, NumHelpers]).
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
 
 
 %% @doc Start the drain FSM, under this supervisor
 -spec start_drain_fsm(proplist()) -> {ok, pid()} | {error, Reason :: term()}.
-start_drain_fsm(CallbackList) ->
+start_drain_fsm(Parameters) ->
+    PartitionToDrain = proplists:get_value(
+        ?DRAIN_PARTITION, Parameters, undefined
+    ),
     supervisor:start_child(
         ?MODULE,
-        {yz_solrq_drain_fsm, {yz_solrq_drain_fsm, start_link, [CallbackList]}, temporary, 5000, worker, []}
+        {PartitionToDrain, {yz_solrq_drain_fsm, start_link, [Parameters]}, temporary, 5000, worker, []}
     ).
-
+-spec start_queue_pair(Index::index_name(), Partition::p()) -> ok.
+start_queue_pair(Index, Partition) ->
+    validate_child_started(
+        supervisor:start_child(?MODULE, queue_pair_spec({Index, Partition}))).
+
+active_queues() ->
+    PairSups = find_pair_supervisors(),
+    PairChildren = lists:flatten([supervisor:which_children(Sup) || Sup <- PairSups]),
+    Workers = [{Index, Partition} || {{worker, Index, Partition}, _Child, _Type, Modules} <- PairChildren, Modules == [yz_solrq_worker]],
```

Comment:
We should break up the super long lines here and in `find_pair_supervisors` below.